projects/plugins/jetpack/extensions/blocks/story/story.php

```diff
@@ -35,89 +35,115 @@ function register_block() {
 add_action( 'init', __NAMESPACE__ . '\register_block' );
 
 /**
- * Add missing `width`, `height`, `srcset` and `sizes` properties to images of the mediaFiles block attributes
+ * Compare 2 urls and return true if they likely correspond to the same resource.
+ * Ignore scheme, ports, query params and hashes and only compare hostname and pathname.
+ *
+ * @param string $url1  - First url used in comparison.
+ * @param string $url2  - Second url used in comparison.
+ *
+ * @returns boolean
+ */
+function is_same_resource( $url1, $url2 ) {
+	$url1_parsed = wp_parse_url( $url1 );
+	$url2_parsed = wp_parse_url( $url2 );
+	return isset( $url1_parsed['host'] ) &&
+		isset( $url2_parsed['host'] ) &&
+		isset( $url1_parsed['path'] ) &&
+		isset( $url2_parsed['path'] ) &&
+		$url1_parsed['host'] === $url2_parsed['host'] &&
+		$url1_parsed['path'] === $url2_parsed['path'];
+}
+
+/**
+ * Enrich media files with extra information we can retrieve from the media library.
+ * Add missing `width`, `height`, `srcset` and `sizes` properties to images of the mediaFiles block attributes.
+ * Add missing `width`, `height`, `poster` properties to videos of the mediaFiles block attributes.
  *
  * @param array $media_files  - List of media, each as an array containing the media attributes.
  *
  * @returns array $media_files
  */
-function with_width_height_srcset_and_sizes( $media_files ) {
-	return array_map(
-		function ( $media_file ) {
-			if ( ! isset( $media_file['id'] ) || ! empty( $media_file['srcset'] ) ) {
-				return $media_file;
-			}
-			$attachment_id = $media_file['id'];
-			if ( 'image' === $media_file['type'] ) {
-				$image = wp_get_attachment_image_src( $attachment_id, 'full', false );
-				if ( ! $image ) {
-					return $media_file;
+function enrich_media_files( $media_files ) {
```

Comment:
In a future PR, I wonder if it may be worth breaking this one down a bit to make it more readable?

Reply:
Yeah, this could be broken down by file type at least: one for videos and one for images

Reply:
Done in https://github.com/Automattic/jetpack/pull/19876/commits/6a3326f4e0190ad4a01fcf09983318e120fe9db9